react/MonthPicker/CustomMonthPicker/CustomMonthPicker.js

```diff
@@ -133,46 +136,60 @@ export default class CustomMonthPicker extends Component {
 
   render() {
     const { value, className, valid, id } = this.props;
+    // eslint-disable-next-line react/prop-types
+    const { label, labelProps, secondaryLabel, tertiaryLabel } = this.props;
+
     const { month, year } = value;
     const monthValue = String(month || '');
     const yearValue = String(year || '');
-
     const rootClasses = classnames({
       [styles.root]: true,
       [className]: className
     });
 
     return (
       <div className={rootClasses}>
-        <Dropdown
-          {...(id ? { id } : {})}
-          options={months}
-          className={styles.dropdown}
-          valid={valid}
-          message={false}
-          placeholder="Month"
-          inputProps={{
-            onBlur: this.handleBlur,
-            onChange: this.handleMonthChange,
-            value: monthValue,
-            className: styles.dropdownInput,
-            ref: this.storeMonthReference
-          }}
-        />
-        <Dropdown
-          options={this.yearOptions}
-          className={styles.dropdown}
-          valid={valid}
-          message={false}
-          placeholder="Year"
-          inputProps={{
-            onBlur: this.handleBlur,
-            onChange: this.handleYearChange,
-            value: yearValue,
-            className: styles.dropdownInput,
-            ref: this.storeYearReference
-          }}
-        />
+        <div className={styles.monthWrapper}>
+          <FieldLabel
+            {...{ id: `${id}-month`, label: label ? <span>{label} <ScreenReaderOnly>Month</ScreenReaderOnly></span> : <ScreenReaderOnly>Month</ScreenReaderOnly>, labelProps, secondaryLabel, tertiaryLabel }}
+          />
```

Comment:
For legibility it would be good to pull these out to internal rendering functions, and maybe shared between both fields if the code remains clear

Reply:
It's different between each field as we only want to pass labelProps, secondaryLabel, tertiaryLabel to the visible label but can pull out to internal functions if you think it's worth it 👍